src/Microsoft.VisualStudio.Composition/ExportProvider.cs

```diff
@@ -1097,12 +1231,33 @@ public object GetValueReadyToRetrieveExportingMembers()
             public void Dispose()
             {
                 this.isDisposed = true;
+
+                if (this.sharingBoundary is null && this.nonSharedPartOwner is null)
```

Comment:
Should we use `this.IsNonShared` instead of `this.sharingBoundary is null` ?

Reply:
Sure. It's equivalent but if you find it easier to read, that's cool.